### PR TITLE
feat(components): create base deck fixture components

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

creates single slot, staging area, waste chute, and waste chute staging area components to be used by the existing BaseDeck component. BaseDeck now receives a deckConfig prop describing a robot deck configuration of fixtures. the deck fixture components render SVGs corresponding to the fixture areas and slot clips provided by each fixture type. positioning and sizing subject to future adjustment.

updates various props in various places to use the `Cutout` type from shared-data for `fixtureLocation`

waste chute fixture foreign object representation is based on a design provided by Rob but not finalized

Screen Shot 2023-10-11 at 3 11 27 PM

closes RAUT-783

# Test Plan

 - verified sizing and layout of fixture components in storybook

# Changelog

 - Creates base deck fixture components

# Review requests

run storybook, take a look

# Risk assessment

low, BaseDeck unused in the app (for now)

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/NotConfiguredModal.tsx [hunk 1]
```diff
@@ -11,16 +11,18 @@ import {
...
 import { LegacyModal } from '../../../../molecules/LegacyModal'
 import { StyledText } from '../../../../atoms/text'
 
+import type { Cutout, FixtureLoadName } from '@opentrons/shared-data'
+
 interface NotConfiguredModalProps {
   onCloseClick: () => void
   requiredFixture: FixtureLoadName
-  cutout: string
+  cutout: Cutout
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/NotConfiguredModal.tsx:25]: oops thanks for fixing this!

File: components/src/hardware-sim/BaseDeck/index.tsx [hunk 2]
```diff
@@ -10,16 +9,33 @@ import {
...
 
+import {
+  // EXTENDED_DECK_CONFIG_FIXTURE,
+  STANDARD_SLOT_DECK_CONFIG_FIXTURE,
+} from './__fixtures__'
+import { SingleSlotFixture } from './SingleSlotFixture'
+import { StagingAreaFixture } from './StagingAreaFixture'
+import { WasteChuteFixture } from './WasteChuteFixture'
+// import { WasteChuteStagingAreaFixture } from './WasteChuteStagingAreaFixture'
+
```
Comment (jerader) [inline components/src/hardware-sim/BaseDeck/index.tsx:33]: wasteChuteStagingAreaFixture to be added later?

File: protocol-designer/src/components/modals/CreateFileWizard/StagingAreaTile.tsx [hunk 3]
```diff
@@ -21,10 +21,10 @@ import { getEnableDeckModification } from '../../../feature-flags/selectors'
 import { GoBack } from './GoBack'
 import { HandleEnter } from './HandleEnter'
 
-import type { DeckConfiguration } from '@opentrons/shared-data'
+import type { Cutout, DeckConfiguration } from '@opentrons/shared-data'
 import type { WizardTileProps } from './types'
 
-const STAGING_AREA_SLOTS = ['A3', 'B3', 'C3', 'D3']
+const STAGING_AREA_SLOTS: Cutout[] = ['A3', 'B3', 'C3', 'D3']
```
Comment (jerader) [inline protocol-designer/src/components/modals/CreateFileWizard/StagingAreaTile.tsx:27]: nice!

**Review by jerader — APPROVED** [review]
WOW it looks so good!!! 💅🤩 can't wait for `BaseDeck` to be wired up throughout the app.

File: app/src/organisms/DeviceDetailsDeckConfiguration/AddDeckConfigurationModal.tsx [hunk 4]
```diff
@@ -28,12 +28,12 @@ import { TertiaryButton } from '../../atoms/buttons'
...
 import { LegacyModal } from '../../molecules/LegacyModal'
 
-import type { FixtureLoadName } from '@opentrons/shared-data'
+import type { Cutout, FixtureLoadName } from '@opentrons/shared-data'
 import type { ModalHeaderBaseProps } from '../../molecules/Modal/types'
 import type { LegacyModalProps } from '../../molecules/LegacyModal'
 
 interface AddDeckConfigurationModalProps {
-  fixtureLocation: string
+  fixtureLocation: Cutout
```
Comment (koji) [inline app/src/organisms/DeviceDetailsDeckConfiguration/AddDeckConfigurationModal.tsx:36]: 👍

File: components/src/hardware-sim/BaseDeck/BaseDeck.stories.tsx [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 import fixture_96_plate from '@opentrons/shared-data/labware/fixtures/2/fixture_96_plate.json'
 import fixture_tiprack_1000_ul from '@opentrons/shared-data/labware/fixtures/2/fixture_tiprack_1000_ul.json'
-import { FLEX_ROBOT_TYPE, LabwareDefinition2 } from '@opentrons/shared-data'
+import {
+  DeckConfiguration,
+  FLEX_ROBOT_TYPE,
+  LabwareDefinition2,
+} from '@opentrons/shared-data'
 
 import type { Meta, StoryObj } from '@storybook/react'
```
Comment (koji) [inline components/src/hardware-sim/BaseDeck/BaseDeck.stories.tsx]: Can we move down `import type`?

File: components/src/hardware-sim/BaseDeck/SlotClip.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react'
+
+export function SlotClip(props: React.SVGProps<SVGPathElement>): JSX.Element {
+  return (
+    <path
+      fill="none"
+      stroke="#16212D"
```
Comment (koji) [inline components/src/hardware-sim/BaseDeck/SlotClip.tsx]: nit
```suggestion
      stroke={COLORS.darkBlackEnabled}
```

**Review by koji — APPROVED** [review]
LGTM

File: components/src/hardware-sim/BaseDeck/index.tsx (hunk 2 above)
Comment (brenthagen) [reply components/src/hardware-sim/BaseDeck/index.tsx:33]: yep, we'll have to add the waste chute staging area as a load name and option in various places also. that all should be a separate ticket i think

**Pull request merged**